pkg/openstack/routes.go

```diff
@@ -288,55 +473,107 @@ func (r *Routes) CreateRoute(ctx context.Context, clusterName string, nameHint s
 
 	klog.V(4).Infof("Using nexthop %v for node %v", addr, route.TargetNode)
 
-	if !r.atomicRoutes {
-		// classical logic
-		r.Lock()
-		defer r.Unlock()
+	mc := metrics.NewMetricContext("router", "get")
+	router, err := routers.Get(r.network, r.os.routeOpts.RouterID).Extract()
```

Comment:
However, I think this is necessary. The `check` step is essential for reconciling. This can ensure CPO can repair a wider range of `route` issues. For example, if a user accidentally and manually deletes a route entity, it can be repaired simply by restarting CPO.